src/views/Upload.vue

```diff
@@ -39,27 +65,131 @@ import UploadTab from '@/components/Upload/UploadTab.vue';
 import UploadFile from '@/components/Upload/UploadFile.vue';
 import UploadText from '@/components/Upload/UploadText.vue';
 
+import axios from 'axios';
+
 export default {
 	components: {
 		Navigation,
 		UploadTab,
 		UploadFile,
-		UploadText,
+		UploadText,	
 	},
 	data: function () {
 		return {
 			selectedTab: '',
 			tabs: ['Nahrať súbory', 'Vložiť text'],
+			files: [],
+			text: '',
+			uploadPercentage: 0,
+			submitted: false,	
+			alert: {
+				message: 'Nezadali ste súbor ani text, ktorý chcete skontrolovať !',
+				type: 'error',
+				duration: 4000,
+			}
 		};
 	},
+	methods: {
+		submitText(){
+			//console.log('text');
+			let formData = new FormData();
+
+			formData.append('file', this.text);
+			formData.append('title','text...');
+
+			if (this.text.length == 0){
+				return this.$store.dispatch('AlertStore/setAlert', this.alert);
+			}
+			axios.post('http://localhost:8000/file/upload/', formData, {
+				headers: {
+					'Content-Type': 'text/plain',
+				},
+				onUploadProgress: function( progressEvent ) {
+					this.submitted = true
+					this.uploadPercentage = parseInt( Math.round( ( progressEvent.loaded * 100 ) / progressEvent.total ) );
+					//console.log(this.uploadPercentage);					
+				}.bind(this)
+			})
+			.then((res) => {
+				console.log('SUCCESS!!' + res);
+			})
+			.then(() =>{
+				this.$store.dispatch('setLoading',false);
+			})
+			.catch(function (e) {
+				console.log('FAILURE!!' + e);
```

Comment:
Maybe reuse the alert, and put the error in there?

Reply:
Well I tried, but it returns such error:
![image](https://user-images.githubusercontent.com/32281312/99788247-21728300-2b21-11eb-8fb4-5d76ec41a95b.png)

Reply:
Yeah, the alert says Unsupported "message" property. "message" msut be string. 

Are you passing any message in there?
You're already using that few lines above.

Usage:
`this.$store.dispatch('AlertStore/setAlert', payload);`

and payload needs to have type, e.g. 'error', and message should be the message in the alert so I think in this case the `e` (error).

Reply:
Okay, I used e.message and it works! Thanks

Reply:
🎉